### PR TITLE
Add TrailingAngleBrackLoc to EditorPlaceholder for upcoming scope code.

### PR DESCRIPTION
**Pull request by davidungar**
Adds an instance variable to track the location of the trailing angle bracket of an editor place holder.
Will be needed by the upcoming scope code. Also adds a test.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

File: lib/Parse/ParseExpr.cpp [hunk 1]
```diff
@@ -2385,6 +2385,7 @@ Expr *Parser::parseExprEditorPlaceholder(Token PlaceholderTok,
   parseTypeForPlaceholder(TyLoc, ExpansionTyR);
   return new (Context) EditorPlaceholderExpr(PlaceholderId,
                                              PlaceholderTok.getLoc(),
+                                             PlaceholderTok.getRange().getEnd(),
```
[the remark this answers is not in the document]
Comment (davidungar) [reply lib/Parse/ParseExpr.cpp]: Thank you--good catch!

File: include/swift/AST/Expr.h [hunk 2]
```diff
@@ -4685,6 +4688,9 @@ class EditorPlaceholderExpr : public Expr {
   SourceRange getSourceRange() const { return Loc; }
   TypeLoc &getTypeLoc() { return PlaceholderTy; }
   TypeLoc getTypeLoc() const { return PlaceholderTy; }
+  SourceLoc getTrailingAngleBracketLoc() const {
+    return TrailingAngleBracketLoc;
```
[the remark this answers is not in the document]
Comment (davidungar) [reply include/swift/AST/Expr.h]: Thank you again, even better idea! I had not thought of it.

**Comment by davidungar** [comment]
@rintaro Thank you so much. I think this version, following your suggestion, is better.

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Review by rintaro — APPROVED** [review]

**Pull request merged**